concent_api/core/tests/test_core_views.py

```diff
@@ -72,3 +74,81 @@ def test_send_should_return_http_400_if_task_id_already_use(self):
         response = self.client.post(reverse('core:send'), data = json.dumps(self.correct_data), content_type = 'application/json', concent_client_public_key = self.public_key, HTTP_CONCENT_CLIENT_PUBLIC_KEY = self.public_key)
 
         self.assertEqual(response.status_code, 400)  # pylint: disable=no-member
+
+
+class CoreViewReceiveTest(TestCase):
+
+    def setUp(self):
+        self.client = Client()
+        self.public_key = '85cZzVjahnRpUBwm0zlNnqTdYom1LF1P1WNShLg17cmhN2UssnPrCjHKTi5susO3wrr/q07eswumbL82b4HgOw=='
+        self.force_data = {
+            "type":      "MessageForceReportComputedTask",
+            "timestamp": 1510911047,
+            "message_task_to_compute": {
+                "type":               "MessageTaskToCompute",
+                "timestamp":          1510909200,
+                "task_id":            1,
+                "deadline":           1510915047
+            }
+        }
+        self.ack_data = {
+            "type":      "MessageAckReportComputedTask",
+            "timestamp": 1510911047,
+            "message_task_to_compute": {
+                "type":               "MessageTaskToCompute",
+                "timestamp":          1510909200,
+                "task_id":            1,
+                "deadline":           1510915047
+            }
+        }
+        self.reject_data = {
+            "type":         "MessageRejectReportComputedTask",
+            "timestamp":    1510911047,
+            "reason":       "cannot-compute-task",
+            "message_cannot_commpute_task": {
+                "type":         "MessageCannotComputeTask",
+                "timestamp":    1510909200,
+                "reason":       "provider-quit",
+                "task_id":      1
+            }
+        }
+
+    @freeze_time("2017-11-17 10:00:00")
+    def test_receive_should_accept_valid_message(self):
+        message_timestamp   = datetime.datetime.now(timezone.utc)
+        new_message         = Message(
+            type        = self.force_data['type'],
+            timestamp   = message_timestamp,
+            data        = json.dumps(self.force_data).encode('utf-8'),
+            task_id     = self.force_data['message_task_to_compute']['task_id']
+        )
+        new_message.save()
+        new_message_status = MessageStatus(
+            message   = Message.objects.get(id = new_message.id),
+            delivered = False
+        )
+        new_message_status.save()
+
+        self.assertEqual(len(MessageStatus.objects.filter(delivered=False)), 1)
+
+        response = self.client.post(
+            reverse('core:receive'),
+            content_type = 'application/json',
+            HTTP_CONCENT_CLIENT_PUBLIC_KEY = self.public_key
+        )
+
+        self.assertEqual(response.status_code, 200)  # pylint: disable=no-member
+        self.assertEqual(len(MessageStatus.objects.filter(delivered=False)), 0)
```

Comment:
You should check that the returned message matches the one that was in the database.